lib/buffer.js

```diff
@@ -66,10 +72,67 @@ function Buffer(arg, encoding) {
   return fromObject(arg);
 }
 
+/**
+ * Functionally equivalent to Buffer(arg, encoding) but throws a TypeError
+ * if value is a number.
+ **/
+Buffer.from = function(value, encoding) {
+  if (typeof value === 'number')
+    throw new TypeError(`'value' argument must not be a number`);
+  return typeof value === 'string' ?
+    fromString(value, encoding) :
+    fromObject(value);
+};
+
 Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype);
 Object.setPrototypeOf(Buffer, Uint8Array);
 
+/**
+ * Creates a new filled Buffer instance.
+ * alloc(size[, fill[, encoding]])
+ **/
+Buffer.alloc = function(size, fill, encoding) {
+  if (typeof size !== 'number')
+    throw new TypeError(`'size' argument must be a number`);
+  size = Math.max(0, size);
+  if (fill !== undefined) {
```

Comment:
If `size === 0` can't we just short circuit here and return early?
```js
size = Math.max(0, size);
if (size === 0)
  return createBuffer(size);
```
This is fine b/c v8 will detect `size === 0` and not call `ArrayBufferAllocator::Allocate`.